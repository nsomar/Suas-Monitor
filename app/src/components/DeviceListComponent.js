import React, { PropTypes } from 'react';

export default class DeviceListComponent extends React.Component { 

    static propTypes = {
        devices: PropTypes.array.isRequired,
        connectToDevice: PropTypes.func.isRequired
    };

    render() {

        let item = (d) => {
            return <li key={d.name + d.type} onClick={() => this.props.connectToDevice(d)}>
                {d.name}
            </li>
        }

        return <ul>
            {this.props.devices.map(d => item(d))}
        </ul>
    }

}
