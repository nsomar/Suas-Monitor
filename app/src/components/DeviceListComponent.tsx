import React = require('react')
import styled from 'styled-components'

const TableRow = styled.tr`
  padding: 4em;
  background: ${props => props.selected ? '#F4F5F6' : 'white' };
`

interface IDeviceListComponentProps {
  iosDevices: Array<any>,
  androidDevices: Array<any>,
  connectToDevice?: ({ }) => {},
  selected: string
}

export default class DeviceListComponent extends React.Component<IDeviceListComponentProps, any> {
  onDeviceSelected = (item: any) => {
    const { data, platform } = item

    if (this.props.connectToDevice) {
      this.props.connectToDevice({
        type: platform === 'Android' ? 'adb' : 'bonjour',
        data
      })
    }
  }

  isSelectedDevice (name) {
    return name === this.props.selected
  }

  render () {
    const { androidDevices, iosDevices } = this.props
    const datasource = [
      ...androidDevices.map(d => ({ name: d.name, platform: 'Android', data: d, selected: this.isSelectedDevice(d.name) })),
      ...iosDevices.map(d => ({ name: d.name, platform: 'iOS', data: d, selected: this.isSelectedDevice(d.name) }))
    ]

    return <div>
      <table>
        <thead>
          <tr>
            <th>Name</th>
            <th>Platform</th>
          </tr>
        </thead>
        <tbody>
          {
            datasource.map((item, i) => <TableRow
              onClick={() => {
                this.onDeviceSelected(item)
              }}
              key={i}
              selected={item.selected}
            >
              <td>{item.name}</td>
              <td>{item.platform}</td>
            </TableRow>)
          }
        </tbody>
      </table>
    </div>
  }
}
