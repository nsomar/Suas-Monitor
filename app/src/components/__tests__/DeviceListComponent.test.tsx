import React = require('react')
import * as renderer from 'react-test-renderer'
import DeviceListComponent from '../DeviceListComponent'

test('renders device list', () => {
  const component = renderer.create(
    <DeviceListComponent
      iosDevices={[]}
      androidDevices={[]}
      selected='test'
    />
  )

  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
