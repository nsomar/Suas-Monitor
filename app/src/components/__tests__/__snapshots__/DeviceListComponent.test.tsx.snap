// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders device list 1`] = `
<div
  className="sc-bwzfXH fIsbOQ"
>
  <div
    className="sc-ifAKCX jXmpmh"
  >
    Applications
  </div>
  <div />
</div>
`;
